### DISCUSSION TITLE
code block language dropdown missing

### DISCUSSION THREAD

**Discussion by knownasilya** [Milkdown/milkdown#942] [Q&A] [answered]
Upgrading to v7 with nord + prism seems to have removed the change language dropdown for code blocks. Was this intentional? Is there a way to get this back?

**Comment by Saul-Mirone** [comment, marked as answer]
Use a node view to do that. See how I implement it in https://github.com/Milkdown/website/blob/main/src/component/Playground/EditorComponent/CodeBlock.tsx

**Reply by knownasilya** [reply]
Is there a non react example of this?

**Reply by knownasilya** [reply]
Would I have to use `$view` https://milkdown.dev/utils#view-t-extends-mark--node-v-extends-nodeviewconstructor--markviewconstructor--getconstructorltttype-t-view-fnctx-ctx--v--view (or async version)?

**Reply by knownasilya** [reply]
Not getting the custom view rendering, nothing shows up in ui or dom:

```ts

const Editor = (props: Props) => {
  const nodeViewFactory = useNodeViewFactory();

  useEditor(root =>
    MilkdownEditor.make()
      .enableInspector()
      .config(nord)
      .config(ctx => {
        ctx.set(rootCtx, root);
        ctx.set(defaultValueCtx, props.markdown);
      })
      .use(commonmark)
      .use(gfm)
      .use(prism)
      .use(emoji)
      .use(indent)
      .use(trailing)
      .use(
        $view(codeBlockSchema.node, () =>
          nodeViewFactory({component: CodeBlock})
        )
      )
  );

  return <Milkdown />;
};
```

I copy pasted you `CodeBlock` implementation from the modified path above.

**Reply by knownasilya** [reply]
Figured it out, needed to wrap the editor in `ProsemirrorAdapterProvider` component.
